rlclientlib/utility/eventhub_http_authorization.h

```diff
@@ -11,17 +12,18 @@ namespace reinforcement_learning {
 
   // The eventhub_client send string data in POST requests to an HTTP endpoint.
   // It handles authorization headers specific for the Azure event hubs.
-  class http_authorization {
+  class eventhub_http_authorization : public i_authorization {
   public:
-    http_authorization(const std::string& host, const std::string& key_name,
+    eventhub_http_authorization(const std::string& host, const std::string& key_name,
       const std::string& key, const std::string& name, i_trace* trace);
-    ~http_authorization() = default;
+    ~eventhub_http_authorization() = default;
     
-    int init(api_status* status);
-    int get(std::string& authorization, api_status* status);
+    virtual int init(api_status* status) override;
+    virtual int get_http_headers(http_headers& headers, api_status* status) override;
 
   private:
     int check_authorization_validity_generate_if_needed(api_status* status);
+    int get(std::string& authorization, api_status* status);
```

Comment:
`get` is not a descriptive enough function name. Please elaborate to describe the functionality.

Reply:
Fixed.